### PR TITLE
fix(stream): fix stream approximate count distinct estimation algorithm.

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Fix stream approximate count distinct estimation bug and add an error ratio test.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#5623

### CODE REVIEW

File: src/stream/src/executor/aggregation/approx_count_distinct.rs [hunk 1]
```diff
@@ -384,6 +384,26 @@ mod tests {
         test_register_bucket_get_and_update_inner::<4>();
     }
 
+    #[test]
+    fn test_error_ratio() {
```
Comment (jon-chuang) [inline src/stream/src/executor/aggregation/approx_count_distinct.rs:388]: Great test. Can we add a comment about the probability that error_ratio < 0.01? Just to make sure we don't get a flakey test.

File: src/expr/src/vector_op/agg/approx_count_distinct.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ const COUNT_BITS: u8 = 64 - INDEX_BITS; // number of non-index bits in each 64-b
 
 // Approximation for bias correction for 16384 registers. See "HyperLogLog: the analysis of a
 // near-optimal cardinality estimation algorithm" by Philippe Flajolet et al.
-const BIAS_CORRECTION: f64 = 0.72125;
+const BIAS_CORRECTION: f64 = 0.7213 / (1. + (1.079 / NUM_OF_REGISTERS as f64));
```
Comment (jon-chuang) [inline src/expr/src/vector_op/agg/approx_count_distinct.rs:31]: Good catch, we will try to be consistent here and use the slightly more accurate formula

File: src/stream/src/executor/aggregation/approx_count_distinct.rs (hunk 1 above)
Comment (jon-chuang) [reply src/stream/src/executor/aggregation/approx_count_distinct.rs:388]: Also, if the `get_max` function is wrong, does this test fail? In other words, does this test act as a regression test for the get_max bug?

Comment (chenzl25) [reply src/stream/src/executor/aggregation/approx_count_distinct.rs:388]: > Also, if the `get_max` function is wrong, does this test fail? In other words, does this test act as a regression test for the get_max bug?

Yes, if we don't fix the `get_max` function, this test will fail.

File: src/expr/src/vector_op/agg/approx_count_distinct.rs (hunk 2 above)
Comment (chenzl25) [reply src/expr/src/vector_op/agg/approx_count_distinct.rs:31]: More importantly, formula can help us change `BIAS_CORRECTION ` when `NUM_OF_REGISTERS ` is changed.

**Review by jon-chuang — APPROVED** [review]
LGTM, thanks for catching this.

**Pull request merged**